koocook_core/tests/test_models.py

```diff
@@ -0,0 +1,608 @@
+from decimal import Decimal
+from typing import List, Union
+import random
+import itertools
+
+from django import test as djangotest
+from django.contrib.auth.models import User
+from django.core.exceptions import ValidationError
+from django.utils import timezone
+
+from koocook_core import models as models_
+from koocook_core.models import *
+from koocook_core.support import Quantity
+from koocook_core.support.fraction import Fraction
+from koocook_core.tests import utils
+
+
+def set_up_authors(n: int = 4) -> List[Author]:
+    test_authors = []
+    for _ in range(n):
+        test_authors.append(Author.objects.create(name=utils.get_full_name()))
+        first_name, last_name = utils.get_first_name(), utils.get_last_name()
+        username = utils.gen_username(first_name, last_name)
+        test_authors.append(
+            User.objects.create(first_name=first_name,
+                                last_name=last_name,
+                                username=username).koocookuser.author)
+    return test_authors
+
+
+def set_up_reviewables(authors: List[Author]
+                       ) -> List[Union[Recipe, Post, Comment]]:
+    test_objects = []
+    for author in authors:
+        recipe = Recipe.objects.create(author=author, name='recipe name')
+        post = Post.objects.create(author=author)
+        test_objects.extend([recipe, post])
+    for obj in random.sample(test_objects, 8):
+        comment = Comment.objects.create(author=random.choice(authors), item_reviewed=obj)
+        test_objects.append(comment)
+        if random.random() > 0.5:
+            comment = Comment.objects.create(author=random.choice(authors), item_reviewed=comment)
+            test_objects.append(comment)
+    return test_objects
```

Comment:
Generally, mock data should not be mutated each time during a test as the test could fail itself instead of the actual implementation. A usage of fixtures is recommended.

Reply:
Actually data can still be mutated with fixtures, but using them would still be very helpful.